adobe/spectrum-web-components
# feat: locale picker in the storybook's decorator

## Description
This PR adds a new control in the storybook decorator that helps the user change the language/locale.
In order to leverage the interaction of `sp-theme` and `LanguageResolutionController` in the storybook it is needed to use a workaround similar to `requestAnimationFrame` so that the story functions get called after the `Theme`'s constructor. 

For more context please check out [this](https://github.com/adobe/spectrum-web-components/pull/4667#discussion_r1713856107) PR comment.
## Motivation and context

Currently if a component is reactive to locale/language changes, in order for the storybook to reflect that it is needed for multiple stories to be created. This doesn't scale as well and it's also cumbersome for users of the SWC storybook to clearly identify the differences between locales.
Currently the most notable behavior change is in the `sp-number-field` component.

## How has this been tested?




    1. Go to `sp-number-field` [storybook](https://imizga-locale-story-decorator--spectrum-web-components.netlify.app/storybook/?path=/story/number-field--default)
    2. Type in a number that has group separators and decimals 
    3. Observe how it is shown with the en-US locale (e.g. 1,000.56)
    4. Change the locale to fr-FR 
    5. Observe the new formatted value (e.g. 1 000,56)
    1. Go to `sp-number-field` [storybook](https://imizga-locale-story-decorator--spectrum-web-components.netlify.app/storybook/?path=/story/number-field--default)
    2. Type in a number that has group separators and decimals 
    3. Observe how it is shown with the en-US locale (e.g. 1,000.56)
    4. Change the locale to de-DE 
    5. Observe the new formatted value (e.g. 1.000,56)


## Screenshots (if appropriate)

<img width="1723" alt="Screenshot 2024-08-23 at 18 03 26" src="https://github.com/user-attachments/assets/444afa96-0643-4f56-be46-bda6cccb22bc">

## Types of changes




## Checklist




## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
